src/server/debug/server/database_dump.go

```diff
@@ -0,0 +1,251 @@
+}
+
+func (s *debugServer) collectDatabaseTableSizes(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "table-sizes", s.queryDatabaseTableSizes, prefix...)
+}
+
+func (s *debugServer) collectDatabaseQuery(ctx context.Context, tw *tar.Writer, fileName string,
+	queryFunc func(ctx context.Context, w io.Writer) error, prefix ...string) error {
+	return collectDebugFile(tw, fileName, "json", func(w io.Writer) error {
+		if err := queryFunc(ctx, w); err != nil {
+			return errors.EnsureStack(err)
+		}
+		return nil
+	}, prefix...)
+}
+
+func (s *debugServer) queryDatabaseRowCounts(ctx context.Context, w io.Writer) error {
+	query := `
+		SELECT schemaname, relname, n_live_tup, seq_scan, idx_scan
+		FROM pg_stat_user_tables
+		ORDER BY schemaname, relname;
+	`
+	return s.writeQueryResponseToJSON(ctx, query, w)
+}
+
+func (s *debugServer) queryDatabaseActivities(ctx context.Context, w io.Writer) error {
+	query := `
+		SELECT current_timestamp - query_start as runtime, datname, usename, client_addr, query
+		FROM pg_stat_activity
+		WHERE state != 'idle'
+		ORDER by runtime DESC;
+	`
+	return s.writeQueryResponseToJSON(ctx, query, w)
+}
+
+func (s *debugServer) queryDatabaseTableSizes(ctx context.Context, w io.Writer) error {
+	query := `
+		SELECT nspname AS "schemaname", relname, pg_total_relation_size(C.oid) AS "total_size"
+		FROM pg_class C
+		  INNER JOIN pg_namespace N ON (N.oid = C.relnamespace)
+		WHERE nspname NOT IN ('pg_catalog', 'information_schema')
+		  AND C.relkind <> 'i'
+		  AND nspname !~ '^pg_toast'
+		ORDER BY nspname, relname;
+	`
+	return s.writeQueryResponseToJSON(ctx, query, w)
+}
+
+func (s *debugServer) collectDatabaseTables(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	tables, err := pachsql.ListTables(ctx, s.database)
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+	for _, table := range tables {
+		table := table // intentional shadow to safely pass by pointer
+		fullPrefix := strings.Join(append(append([]string{}, prefix...), "tables", table.SchemaName), "/")
+		if err := collectDebugFile(tw, table.TableName, "json", func(w io.Writer) error {
+			if err := s.collectTable(ctx, w, &table); err != nil {
+				return errors.EnsureStack(err)
+			}
+			return nil
+		}, fullPrefix); err != nil {
+			return errors.EnsureStack(err)
+		}
+	}
+	return nil
+}
+
+func (s *debugServer) collectTable(ctx context.Context, w io.Writer, table *pachsql.SchemaTable) error {
+	column, err := pachsql.GetMainColumn(ctx, s.database, table)
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+	rows, err := s.getFirstPage(ctx, column, pageSize, table)
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+	if len(rows) == 0 {
+		return writeAllRowsToJSON(rows, w)
+	} else if err := writeRowPageToJSON(rows, w, firstPage); err != nil {
+		return errors.EnsureStack(err)
+	}
+	queryTemplate := fmt.Sprintf(`
+		SELECT * FROM %s.%s WHERE %s > {{lastitem}} 
```

Comment:
this assumes that the WHERE column is unique. Is that a fair assumption?

Reply:
Yes it does assume that. The assumption should hold up if our KeyColumn() is unique. I _believe_ it is given the tables we have today